source/common/network/filter_state_dst_address.h

```diff
@@ -22,6 +22,7 @@ class DestinationAddress : public StreamInfo::FilterState::Object {
 
 private:
   const Network::Address::InstanceConstSharedPtr address_;
+  friend class DestinationAddressReflection;
```

Comment:
how do you see further supported reflection objects being implemented ? will we just propagate friend classes for every reflection added?

Reply:
It can be done without friends, via public API. But in general, a friend seems appropriate for reflection support.